include/vicon/udp.h

```diff
@@ -286,61 +315,65 @@ class UDPClient
         m_ReadThread = std::thread(&UDPClient::readThread, this, socket);
     }
 
-    unsigned int getNumObjects()
+    size_t getNumObjects()
     {
-        std::lock_guard<std::mutex> guard(m_ObjectDataMutex);
+        waitUntilConnected();
+        std::lock_guard<std::mutex> guard(m_ObjectMutex);
         return m_ObjectData.size();
     }
 
-    unsigned int findObjectID(const std::string &name)
-    {
-        // Search for object with name
-        std::lock_guard<std::mutex> guard(m_ObjectDataMutex);
-        auto objIter = std::find_if(m_ObjectData.cbegin(), m_ObjectData.cend(),
-            [&name](const ObjectDataType &object)
-            {
-                return (strcmp(object.getName(), name.c_str()) == 0);
-            });
-
-        // If object wasn't found, raise error
-        if(objIter == m_ObjectData.cend()) {
-            throw std::out_of_range("Cannot find object '" + name + "'");
-        }
-        // Otherwise, return its index i.e. object ID
-        else {
-            return (unsigned int)std::distance(m_ObjectData.cbegin(), objIter);
-        }
-    }
-
     //! Get current pose information for specified object
-    ObjectDataType getObjectData(unsigned int id)
+    ObjectDataType getObjectData(const std::string &name)
     {
-        std::lock_guard<std::mutex> guard(m_ObjectDataMutex);
-        return m_ObjectData.at(id);
+        waitUntilConnected();
+        std::lock_guard<std::mutex> guard(m_ObjectMutex);
+        return m_ObjectData.at(name);
     }
 
+     //! Get current pose information for first object
+     ObjectDataType getObjectData()
+     {
+         waitUntilConnected();
+         std::lock_guard<std::mutex> guard(m_ObjectMutex);
+         return m_ObjectData.begin()->second;
+     }
+
+     auto getObjectReference(Stopwatch::Duration timeoutDuration = 10s) const
+     {
+         waitUntilConnected();
+         std::lock_guard<std::mutex> guard(m_ObjectMutex);
+         return ObjectReference<ObjectDataType>(*this,
+                                                m_ObjectData.begin()->first,
+                                                m_ObjectData.begin()->second,
+                                                timeoutDuration);
+     }
+
     //! Returns an object whose pose is updated by the Vicon system over time
-    auto getObjectReference(unsigned int id,
-                            Stopwatch::Duration timeoutDuration = Stopwatch::Duration::max())
+    auto getObjectReference(const std::string& name,
+                            Stopwatch::Duration timeoutDuration = 10s) const
     {
-        return ObjectReference<ObjectDataType>(*this, id, timeoutDuration);
+        return ObjectReference<ObjectDataType>(*this,
```

Comment:
Shouldn't this version also include:
```c++
 waitUntilConnected();
std::lock_guard<std::mutex> guard(m_ObjectMutex);
```
?

Reply:
Not if we get rid of the unnecessary use of ``m_ObjectData`` here.